### ISSUE TITLE
Reconnect - dialogues and errors appearing repeatedly despite reconnection

### ISSUE THREAD

**Issue by gerbrent** [nextcloud/text#5134]
**Describe the bug**
* The reconnect dialogue appears if, say, a document has been in edit mode and one's computer had been off for an extended period of time.
* Then, if `Reconnect` button is clicked, the system appears to reconnect. 
* Only after a few seconds, connection errors appear (`Connection failed.`) and the reconnect button reappears, only to vanish again and repeat the cycle indefinitely every 3-4 seconds.

May be related to #4943 ?

**To Reproduce**
1. open and .md document for editing (in browser).
2. Keep the doc open, the move offline for extended period of time. (i.e. computer sleep)
3. Move online (i.e. unsleep/wake computer)
4. See Reconnect error w Reconnect button (expected behaviour!)
5. Click Reconnect button
6. Have error messages and Reconnect button reappear every 3-4 seconds indefinitely.

**Expected behavior**
Editor reconnects, and errors do not persist/reappear. 

**Screenshots**
![Peek 2023-12-15 14-34 - edit gif](https://github.com/nextcloud/text/assets/7018928/f83d4c78-5774-43ee-91ee-5e6250810af6)

**Server details:**

- Nextcloud version: 28.0.0
- PHP Version: 🤷
- Database: 🤷

**Client details:**
 - OS: NixOS
 - Browser: Firefox 120.0.1
 - Browser version: 120.0.1
 - Device: desktop

<details>
<summary>Logs</summary>

#### Browser log
```
Uncaught (in promise) Error: Close has already been called on the connection
    C SessionApi.js:28
    I SessionApi.js:183
    close SessionApi.js:176
    _close SyncService.js:295
    close SyncService.js:287
    close WebSocketPolyfill.js:120
    _checkInterval y-websocket.js:374
    r NextJS
SessionApi.js:28:2
[ERROR] text: Unexpected Error "Failed to apply steps. Retry!"
Object { app: "text", uid: "brent", level: 2, error: Error }
sentry.js:2:63997
Uncaught (in promise) Error: Close has already been called on the connection
    C SessionApi.js:28
    I SessionApi.js:183
    close SessionApi.js:176
    _close SyncService.js:295
    close SyncService.js:287
    close WebSocketPolyfill.js:120
    _checkInterval y-websocket.js:374
    r NextJS
SessionApi.js:28:2
Uncaught (in promise) Error: Close has already been called on the connection
    C SessionApi.js:28
    I SessionApi.js:183
    close SessionApi.js:176
    _close SyncService.js:295
    close SyncService.js:287
    close WebSocketPolyfill.js:120
    _checkInterval y-websocket.js:374
    r NextJS
2 SessionApi.js:28:2
Got notification data, restoring default polling interval. sentry.js:2:63997
Uncaught (in promise) Error: Close has already been called on the connection
    C SessionApi.js:28
    I SessionApi.js:183
    close SessionApi.js:176
    _close SyncService.js:295
    close SyncService.js:287
    close WebSocketPolyfill.js:120
    _checkInterval y-websocket.js:374
    r NextJS
```

</details>

**Comment by juliushaertl** [comment]
Likely the patch mentioned in https://github.com/nextcloud/text/pull/5126#discussion_r1427739887 could help here

**Comment by AdamPS** [comment]
I am seeing something that looks very much like this. I see these errors repeating in the browser console.

- [ERROR] text: Unexpected Error "Failed to apply steps. Retry!"
- Uncaught (in promise) Error: Close has already been called on the connection

In the worst case the app is unusable.

**Comment by AdamPS** [comment]
Workaround: hard-refresh the browser (Ctrl+F5) or start a new tab/window

**Comment by nicofrand** [comment]
I get something similar with the android app and the WebView (rich editor), after switching between apps or after some idle time.

**Comment by mejo-** [comment]
This should have been fixed by https://github.com/nextcloud/text/pull/5126. The fix will get released with the upcoming 26.0.11, 27.1.6 and 28.0.2 releases. Please comment and reopen if you still discover the issues after upgrading to one of these versions.
